### DISCUSSION TITLE
Live log streaming not working

### DISCUSSION THREAD

**Discussion by elcajon** [homeassistant-apps/app-cloudflared#744] [Q&A] [answered]
@brenner-tobias 
I'm not sure if it's a problem on my end, but since the introduction of the [live log](https://www.home-assistant.io/blog/2024/11/06/release-202411/#live-logs) with the last core/supervisor release, I no longer get the current logs displayed. When I access HA locally (using my internal IP) it works. Is this a problem with my (remotely managed) tunnel or are you seeing this on your end as well? I am not sure as there are no reported issues in this repository so far. I've experimented a bit with tunnel run parameters and cloudflare caching options, but can't seem to solve the problem.

For reference:

latest supervisor logs displayed when viewing with Cloudflare domain:

<img width="1367" alt="image" src="https://github.com/user-attachments/assets/36f99fd4-e356-426b-9a43-caec3bba6d26">

and checking with local ip (xxx:8123):

<img width="1166" alt="image" src="https://github.com/user-attachments/assets/b9427e94-6cb2-40ea-aa46-8840d13a9656">

I guess it is related to this connection, but not sure....

<img width="1433" alt="image" src="https://github.com/user-attachments/assets/ceb344bb-68b8-4a14-9082-e44978ac4afd">

**Comment by brenner-tobias** [comment]
Thanks for raising, I did not realise it until now, but I do have the same issue on my end on all my add-ons.
I can scroll-up to load older entries, but no new entries arrive.

**Comment by brenner-tobias** [comment]
@AlecRust, @elcajon, @felipecrs any ideas, what might be causing this?

**Reply by felipecrs** [reply]
I once had a somewhat similar issue: https://github.com/esphome/issues/issues/4989

It was fixed by: https://github.com/home-assistant/addons/pull/3356

This is everything I thought of, so far.

**Reply by elcajon** [reply]
Unfortunately I have no idea, I suspect it is due to the Cloduflare proxy filtering the live logs.
It would be interesting to test:
- The new camera WebRTC feature works through the tunnel, or if there are problems there too. I suspect that the protocol is also used for the logs, but I'm not sure. We would have to check the changes in the supervisor/frontend.
- Live logging via a Cloudflare proxy domain without a tunnel works (port sharing and nginx proxy for example)

**Reply by elcajon** [reply]
> The new camera WebRTC feature works through the tunnel, or if there are problems there too. I suspect that the protocol is also used for the logs, but I'm not sure. We would have to check the changes in the supervisor/frontend.


I am wrong with this, as I learned here: https://github.com/home-assistant/supervisor/issues/5509#issuecomment-2565518797

> Live logging via a Cloudflare proxy domain without a tunnel works (port sharing and nginx proxy for example)

I did a test and realized that live logs indeed do work when using Nginx Proxy add-on with Cloudflare DNS and activated Cloudflare DNS proxy --> So the problem is directly related to the cloudflared (tunnel)

**Reply by felipecrs** [reply]
EDIT: Never mind. You meant using the Nginx add-on standalone, without cloudflared.

~What do you mean exactly by _Nginx Proxy add-on with Cloudflare DNS_? Do you mean enabling this option?~

![image](https://github.com/user-attachments/assets/6615bedd-e15f-4bfa-85df-dbb578bce570)

**Reply by othorp** [reply]
I described a potential workaround to still use cloudflared tunnel by using nginx to rewrite response headers on the thread at https://github.com/home-assistant/supervisor/issues/5509

**Comment by elcajon** [comment]
As of [v5.2.8](https://github.com/brenner-tobias/addon-cloudflared/releases/tag/untagged-923d63d6e3416a082465) (not released at the time of writing) you can change your [**remote managed tunnel**](https://github.com/brenner-tobias/addon-cloudflared/wiki/How-tos#how-to-configure-remote-tunnels) to get log streaming working again by changing your service uri from `homeassistant:8123` to `localhost:8321`. Feel free to share your feedback after the version has been released.

<img width="1470" alt="image" src="https://github.com/user-attachments/assets/386c5e56-529e-4101-a5c9-de412a7c8f56" />

**Reply by brenner-tobias** [reply]
you can also switch to the [edge repository](https://github.com/brenner-tobias/ha-addons-edge), where it is already included

**Reply by felipecrs** [reply]
An option to override the default Home Assistant address would be nice too, for users that are not using the Cloudflare Dashboard.

For example, I already use Nginx Proxy Manager. I could just override `homeassistant:8123` to `a0d7b954-nginxproxymanager` instead.

**Comment by brenner-tobias** [comment]
I am running this since yesterday and it seems like it is not working 100% @elcajon.
Right now, I am missing logs in my view via the tunnel since yesterday 22:52, while I do have around 19 new lines when checking locally. 
How does it look for you?

**Reply by elcajon** [reply]
I had looked at various add-on logs a few times today and compared each between Cloudflare domain and local domain and it all seemed to look good.

But it looks like an upstream correction is currently in the works, so I would say we'll wait and see what it looks like afterwards, maybe our workaround is no longer necessary... fingers crossed

@felipecrs  I would suggest to wait extending the workaround and wait a little longer to see if the problem can be solved upstream

https://github.com/home-assistant/supervisor/issues/5509#issuecomment-2587547343
https://github.com/home-assistant/supervisor/pull/5544

**Reply by felipecrs** [reply]
I totally agree.

**Reply by elcajon** [reply]
Have any of you tested the new Supervisor version (recently released in beta)? For me, nothing seems to have changed for the better.

**Reply by othorp** [reply]
The supervisor update only adds the X-Accel-Buffering header which should improve nginx behaviour but unlikely to fix cloudflared.

Full-fix likely to be more than trivial header rewrite and might require rewriting home-assistant live logging to use Server Side Events. 
While the misuse of the mime-type here seems to resolve the problem, the behaviour is not fully specified and may break depending on the presence of colons and newlines in the log data.  In addition, without structuring the data correctly the home assistant server won't be sending keep-alives to keep the connection open.

**Reply by elcajon** [reply]
Do you have any recommendations on how we should deal with this from an add-on perspective?
Unfortunately, I'm not deep enough into this proxy topic to judge whether there's anything we can do from an add-on perspective or not. 
I'm also wondering if this means we should use the recently introduced nginx workaround as the new default for the HomeAssistant-Cloudflare tunnel, or if there's a better alternative.
I'm also not sure whether our workaround can be improved in any way.

**Reply by othorp** [reply]
From my point of view, the workaround as-is improves things (given that without this log streaming doesn't work at all).  Therefore it is worth consideration for inclusion as default.  API interference is only to logging API endpoints so shouldn't have wider impact.  I think it's possible to restrict this further to pick up just the "follow" API endpoints with:

`location ~ /api/hassio/.*/logs.*/follow`

**Remaining Issues to consider:**

1. The nginx "follow" API calls timeout after 60s (nginx proxy_read_timeout).  This issue is also there with the main nginx addon and results in the home-assistant frontend repeating log lines **(TODO: Raise home assistant issue - lack of keep-alive)**
2. Mapping content-type to text/event-stream does change browser behaviour as it expects SSE EventStream.  Most browsers limit simultaneous SSE connections to 6.  At least in Edge / Chrome I can see that that home-assistant live logs stop once they timeout more than 6 times - the SSE connections don't seem to be tidied up properly in the browser post timeout.  I don't think this can be improved directly in the addon, although I think it's possible to avoid this by rewriting content-type back to text/plain via CloudFlare Dashboard Page Rules.

**Reply by elcajon** [reply]
Thanks for the suggestion, I changed the nginx location to match only the "follow" endpoint.
@brenner-tobias I would suggest to release a new version after #780 is merged, maybe with a comment to this discussion.

**Comment by brenner-tobias** [comment]
@elcajon I think we can release the new updates. What do you think? If you agree, feel free to do so

**Comment by felipecrs** [comment]
I can confirm live logs work like a charm now!

**Comment by elcajon** [comment]
I did notice some performance issues using the new proxy while not connected to home network (e.g. metered connection, work, ...). Looking at the latest [community posts](https://community.home-assistant.io/t/new-add-on-cloudflared/361637/721) I am not the only one but it is also not a general problem. 
I will add some more proxy settings (also used by hassio-addons), hopefully this will improve performance.

**Comment by tzagim** [comment]
The log doesn't appear for me in version 5.3.1.

When I download the log I see the links it tries to access and they are invalid.
```
ERR  error="context canceled" connIndex=3 event=1 ingressRule=0 originService=http://homeassistant:8123
ERR Request failed error="context canceled" connIndex=3 dest=https://MyHomeAssistantURL/api/hassio/addons/855ps6_cloudflared/logs/follow?lines=100 event=0 ip=198.41.200.0 type=http
```

I'm throwing out information that I'm not sure is related to the problem.

This is the address range of Cloudflare:
```yml
198.41.192.0/24
198.41.200.0/24
```
No log appears in any of the extensions, only when I connect using a local IP address do I immediately receive the log.

**Reply by felipecrs** [reply]
> No log appears in any of the extensions

To be sure, do you mean the add-ons?

**Reply by tzagim** [reply]
@felipecrs 
Yap, and now I see that system logs cannot be viewed either.

Until now I was using a local tunnel and everything worked fine except for problems connecting with the mobile app in my tablet :( So I switched to a remote tunnel.
Since the switch, i cen connecting with the mobile app in my tablet BUT logs cannot be seen.

**Reply by elcajon** [reply]
This will work if you change the destination URL from http://homeassistant:8123 to http://localhost:8321

**Reply by tzagim** [reply]
Don't know why I changed it during the migration to remote control. I guess I wanted to stick to the installation guide.
Now it really works like a charm.
Thank you very much @elcajon

**Reply by felipecrs** [reply]
It's weird though that local tunnel didn't work for you, but remote tunnel did. I have been using local tunnel since ever.

**Reply by tzagim** [reply]
I think it has something to do with SSL. For some reason it didn't recognize the local security certificate, so I couldn't connect.

After switching to a remote tunnel, it recognized the Cloudflare certificate and allowed me to connect normally from the tablet as well.

**Comment by dm82m** [comment]
hey guys, I have the same problem that I do not get any logs for addons if I access my instance via domain name. If I access it locally with the IP it works.

My setup is as following:
- Cloudflare tunnel with this addon here
- Catchall for Nginx Proxy Manager
- Cloudflare is doing SSL
- Nginx Proxy Manager is distributing everything via http to the local services

I read a lot here (and in home assistant repo) like switching from local to remote tunnel, setting urls in the tunnel, removing proxy of the cloudflare dns entry, adding something to the nginx proxy manager to disable caching and so on. but honestly I lost track of what is really needed to be done to get the addon logs back working with the dns setup.

Any help appreciated!

Best,
Dirk

---

edit: okay I found something interesting: first the log says "protocol is being loaded" and 1-2 secs later goes to "No errors were reported." and if I just wait couple of minutes then the log is displayed. It feels like it is only loaded if a new line would be added to it. if that is not the case the message mentioned before is visible.

edit2: but on my companion app on ios it just gets stuck with "protocol is being loaded ..." and nothing more happens.

edit3: and sometimes (on mac and ios) I get a 504 gateway timeout html error code within the log view...

**Reply by elcajon** [reply]
@dm82m This could be related to #892 do you run the tunnel local or remote managed? If it is remote managed, make sure to change the proxy URL to `127.0.0.1:8321`. 

just re-read your post. I guess it is locally managed and Nginx Catch all is enabled, right? Do you use NPM to access Home Assistant or the add-on configured `external_hostname`?

**Reply by dm82m** [reply]
Yes it’s local and nginx catch all is activated (I am using more endpoints through the tunnel), so nginx is used to map to home assistant

**Reply by dm82m** [reply]
```
external_hostname: ""
additional_hosts: []
use_builtin_proxy: false
nginx_proxy_manager: true

```

**Reply by elcajon** [reply]
If you want to access HA through NPM, you need to add something like this to your nginx proxy options. (I’m not sure if that’s possible since I don’t use NPM.) Alternatively, you could remove the HA Host entry from NPM and use our `external_hostname` option to access your HA Host.

https://github.com/brenner-tobias/addon-cloudflared/blob/7372b59321c8cf2f07a800e59d315fa484440519/cloudflared/rootfs/etc/nginx/template/homeassistant.gtpl#L32-L35

**Reply by dm82m** [reply]
Will catch all work for the rest? So external_url for HA and catch all for the rest to npm?

**Reply by elcajon** [reply]
Yes, exactly
https://github.com/brenner-tobias/addon-cloudflared/blob/main/cloudflared/DOCS.md#option-nginx_proxy_manager

**Reply by dm82m** [reply]
Okay it works now, thank you for your help.

both scenarios are working:
1) With this addon and external_url and use_builtin_proxy

2) and also with npm and the lines mentioned by @elcajon 

again: thanks for your help!

**Comment by tomballgithub** [comment]
While trying to debug this issue, I removed this addon from HA. I previously had it installed, but I was also running the service on my server and remotely managing it. I read the docs and searched, but I was unable to determine what this add-on does above/beyond running the service outside of HA. Is it simply that it's yet another way to install and manage a Cloudflare tunnel, and it has some additional configuration options that allow a fix for the logging issue? The tips to fix this don't seem to apply to a standard remotely-managed configuration. If I want to run this within HA, is it OK to have both instances running, or do I need to turn off my original remotely-managed service?

**Comment by tomballgithub** [comment]
Well, I solved this on my own, FWIW. I am using a remotely managed cloudflare tunnel running on my server. I am NOT using the HA addins for Cloudflared or NGINX. I did install NGINX (NPM) on my server, listening on port 8124. I redirected my domain to port 8124, which NGINX was listening on. And I used the following minimal config, leveraged off what was provided above and tweaked by Chat-GPT5. No other services using the tunnel are using NGINX. Logs are working great!

```
events {}

http {
    server {
        # Cloudflare Tunnel will connect here internally
        listen 8124;
        server_name ha2.krontz.com;

        # --- WebSocket endpoint ---
        location /api/websocket {
            proxy_pass http://192.168.1.86:8123/api/websocket;
            proxy_http_version 1.1;

            # Force upgrade headers (no map needed)
            proxy_set_header Upgrade websocket;
            proxy_set_header Connection Upgrade;

            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;

            proxy_read_timeout 3600;
            proxy_buffering off;
        }

        # --- Everything else (UI, API calls, static files) ---
        location / {
            proxy_pass http://192.168.1.86:8123;
            proxy_http_version 1.1;

            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
        }

		location ~ ^/api/hassio/.*/logs {
            proxy_pass http://192.168.1.86:8123;
            proxy_http_version 1.1;
			proxy_set_header Connection '';
			proxy_set_header Host $host;

			proxy_buffering off;
			proxy_cache off;
			proxy_read_timeout 3600;

			# Preserve SSE headers
			proxy_set_header Accept text/event-stream;
			proxy_hide_header Content-Type;
			add_header Content-Type "text/event-stream";
        }
    }
}
```

**Reply by felipecrs** [reply]
That's the same as using this add-on with remotely managed tunnel and pointing the Home Assistant service URL in Cloudflare to http://localhost:8321.

It's just harder because you need to manage NGINX configuration by yourself.

**Comment by felipecrs** [comment, marked as answer]
For everyone in this thread:

It looks like the workaround is no longer needed, and therefore we have removed the built-in NGINX from the latest Edge version:

- https://github.com/brenner-tobias/addon-cloudflared/pull/913

Everyone that had switched the service URL from `homeassistant:8123` to `localhost:8321` need to switch again.

You can already switch even if you don't have the latest Edge version installed, and then let us know if you notice any issue with live logs.

**Reply by dm82m** [reply]
Is the solution also working in case one has done the additional npm proxy path entry and not used builtin proxy of the addon?

**Reply by felipecrs** [reply]
Hopefully. :)

**Reply by dm82m** [reply]
Can confirm it works on my end

**Reply by Partizanin** [reply]
Cloud someone tell me how to switch HA port in cloudflare addon config?

**Reply by brenner-tobias** [reply]
It depends on how you configured the add-on. If you have changed the port in a remote managed tunnel, you have to change it in the Cloudflare Zero Trust Dashboard.
If you changed it locally, you have to change it in the add-on configuration yaml. In any case, if the add-on is still functioning after the update, you do not have to change anything.

**Reply by Partizanin** [reply]
> It depends on how you configured the add-on. If you have changed the port in a remote managed tunnel, you have to change it in the Cloudflare Zero Trust Dashboard. If you changed it locally, you have to change it in the add-on configuration yaml. In any case, if the add-on is still functioning after the update, you do not have to change anything.

For the truth, i dont remember how i configure CF tunnel, just update my addon and after that it do not work anymore.
I have this log error 
<img width="1677" height="59" alt="image" src="https://github.com/user-attachments/assets/8bdb2bd5-0e2d-4d02-b0c5-2d548a20bb34" />


I have changed port from 8321 to 8123 in cloudflare tunnel rout config on the web but it didnt help

**Reply by brenner-tobias** [reply]
So you are using a remote managed tunnel? In that case it seems like something went wrong configuring the remote tunnel. In the Cloudflare Zero Trust Dashboard you have to navigate to "Network-Tunnel", click on your tunnel name and then select "Edit" in the right menu. After that, you have to jump to published Application routes, select your published (sub) domain and change the service to "http://homeassistant:8123".

**Reply by Partizanin** [reply]
> So you are using a remote managed tunnel? In that case it seems like something went wrong configuring the remote tunnel. In the Cloudflare Zero Trust Dashboard you have to navigate to "Network-Tunnel", click on your tunnel name and then select "Edit" in the right menu. After that, you have to jump to published Application routes, select your published (sub) domain and change the service to "http://homeassistant:8123".

It already with such config
 _http://homeassistant:8123_ and it was working with cloudflare addon 5.3.9

After update addon to 6.0.0 it doesn't work with port 8123 and 8321 also

**Reply by brenner-tobias** [reply]
There must be something else wrong in your configuration. I suggest to uninstall the add-on and do the set-up from scratch. If there are still issues, please have a look at our open issues in this repository or open one yourself.

**Reply by Partizanin** [reply]
> There must be something else wrong in your configuration. I suggest to uninstall the add-on and do the set-up from scratch. If there are still issues, please have a look at our open issues in this repository or open one yourself.

yes, it helped, thanks, totally reinstalled addon with container data and HA was reloaded, also was removed existed tunnel on cloudflare

**Reply by tobi162** [reply]
Hi @brenner-tobias,
suddently the remote access stopped working on my phone.
Browser and other phones using the app with the cloudflare connection are still working and dont have any issues.
My Samsung Galaxy S22 has the error message on the screen "certificate athority is not trusted". Nothing helped and i have already invested several ours in changing cloudflare settings, restarting and reinstalling HA addon, requesting new certificates, …

Would be great if you have some hint what i did wrong here? I used the default way to configure the addon. No special settings.

Thanks!

**Comment by tomballgithub** [comment]
Why is it no longer needed? I read the recent posts here and the referenced change #913, and don't see it addressed. Did something change on the Home Assistant side?

**Reply by felipecrs** [reply]
I have no clue what changed and where, but I'm no longer able to reproduce the issue. Did you observe something different?

**Reply by othorp** [reply]
Suspect this could have been fixed by a change in Cloudflare behaviour to respect cache-control headers by default on the free plans:
https://developers.cloudflare.com/cache/concepts/default-cache-behavior/

I can't see anything in the Cloudflare changelog but I remember the default behaviour being disabled when I first looked at this.

**Reply by felipecrs** [reply]
Very likely to be a change in Cloudflare side.

**Reply by tomballgithub** [reply]
I had a Cloudflare tunnel public hostname still pointing to my Home Assistant outside of NGINX. I verified that the issues I observed with logs and also websockets all seem to work correctly. This is great news!
